src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -196,16 +193,31 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
             {
                 // We are on the wrong thread.
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer did not provide Edit Access"));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
             if (subjectBuffer.EditInProgress)
             {
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer is editing by someone else."));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
+            }
+
+            CompletionChange change;
+
+            // We met an issue when external code threw an OperationCanceledException and the cancellationToken is not cancelled.
+            // Catching this scenario for further investigations.
+            try
+            {
+                change = completionService.GetChangeAsync(document, roslynItem, completionListSpan, commitCharacter, cancellationToken).WaitAndGetResult(cancellationToken);
+            }
+            catch (OperationCanceledException e) when (e.CancellationToken != cancellationToken)
```

Comment:
when does this happen?

Reply:
We do not know. @heejaechang got a request from a customer. We reviewed our code and could not find any suspicious throws. That is why, we want to get a dump.